internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/router/TestRead.java

```diff
@@ -121,11 +158,56 @@ protected VeniceClusterWrapper getVeniceCluster() {
     return veniceCluster;
   }
 
-  @BeforeClass(alwaysRun = true)
-  public void setUp() throws VeniceClientException, ExecutionException, InterruptedException {
+  @BeforeMethod
+  public void setUp(Method method) throws VeniceClientException, ExecutionException, InterruptedException {
     if (!isTestEnabled()) {
       return;
     }
+    if ("testBlobDiscovery".equals(method.getName())) {
+      setUpForBlobTransfer();
+    } else {
+      setUpTestRead();
+    }
+  }
+
+  public void setUpForBlobTransfer() {
```

Comment:
The idea of setting up totally different things for a test case in a single test file doesn't sound great to me. If a new setup is required, let's move to a new integ test file instead of complicating this file.

Reply:
created a new file